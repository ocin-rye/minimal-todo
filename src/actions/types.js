// Action Types
export const ADD_TODO = 'add_todo';
export const DELETE_TODO = 'delete_todo';
export const SET_TODO_LIST_FILTER = 'set_todo_list_filter';
export const TOGGLE_TODO_COMPLETION = 'toggle_todo_completion';
