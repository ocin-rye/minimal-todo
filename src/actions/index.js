import { ADD_TODO, DELETE_TODO, SET_TODO_LIST_FILTER, TOGGLE_TODO_COMPLETION } from './types';

// Actions Creators - Mock Async Functionality
export const addTodo = (todoText, todoId) => {
  const action = {
    type: ADD_TODO,
    payload: {
      text: todoText,
      completed: false,
      id: todoId,
    },
  };

  return action;
};

export const deleteTodo = todo => {
  const action = {
    type: DELETE_TODO,
    payload: todo,
  };

  return action;
};

export const toggleTodoCompletion = todo => {
  const action = {
    type: TOGGLE_TODO_COMPLETION,
    payload: todo,
  };

  return action;
};

export const setTodoListFilter = filterSetting => {
  const action = {
    type: SET_TODO_LIST_FILTER,
    payload: {
      todoListFilter: filterSetting,
    },
  };

  return action;
};
