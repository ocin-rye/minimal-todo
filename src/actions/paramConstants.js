// Action Creator Parameter Constants:
// A list of constants for action creators that only accept
// a predefined set of parameters.
// Also used with functions within react comnponents related to action creators.

// setTodoListFilter
export const todoListFilters = {
  SHOW_ALL: 'All',
  SHOW_COMPLETED: 'Completed',
  SHOW_ACTIVE: 'Active',
};
