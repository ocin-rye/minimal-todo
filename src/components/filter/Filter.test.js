import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Filter from './index';
import { todoListFilters } from '../../actions/paramConstants';

describe('<Filter />', () => {
  const mockSetFilter = jest.fn();
  const filterState = 'All';
  const filter = shallow(<Filter setTodoListFilter={mockSetFilter} todoListFilter={filterState} />);

  it('renders without crashing', () => {
    expect(toJson(filter)).toMatchSnapshot();
  });

  it('renders filter buttons', () => {
    const buttonLength = Object.keys(todoListFilters).length;

    expect(filter.find('button')).toHaveLength(buttonLength);
  });

  describe('set filter', () => {
    it('calls filter fn with all param', () => {
      filter.find('.All').simulate('click');
      expect(mockSetFilter).toBeCalledWith('All');
    });

    it('calls filter fn with active param', () => {
      filter.find('.Active').simulate('click');
      expect(mockSetFilter).toBeCalledWith('Active');
    });

    it('calls filter fn with completed param', () => {
      filter.find('.Completed').simulate('click');
      expect(mockSetFilter).toBeCalledWith('Completed');
    });

    // Generate click fn call test for each filter button
    // Object.keys(todoListFilters).map(key => {
    //   it(`calls filter fn with ${todoListFilters[key]} param`, () => {
    //     filter.find(`.${todoListFilters[key]}`).simulate('click');
    //     expect(mockSetFilter).toBeCalledWith('todoListFilters[key]');
    //   });
    // });
  });
});
