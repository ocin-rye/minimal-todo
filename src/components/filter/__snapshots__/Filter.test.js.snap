// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> renders without crashing 1`] = `
<div
  className="container"
>
  <p>
    Filter Todos
  </p>
  <button
    className="button buttonActivated All"
    key="filter button All"
    onClick={[Function]}
  >
    All
  </button>
  <button
    className="button null Completed"
    key="filter button Completed"
    onClick={[Function]}
  >
    Completed
  </button>
  <button
    className="button null Active"
    key="filter button Active"
    onClick={[Function]}
  >
    Active
  </button>
</div>
`;
